packages/@aws-cdk/aws-rds/lib/props.ts

```diff
@@ -97,27 +97,142 @@ export interface BackupProps {
 /**
  * Username and password combination
  */
-export interface Login {
+export abstract class Login {
+
+  /**
+   * Creates a Login for the given username, and optional password and key.
+   * If no password is provided, one will be generated and stored in SecretsManager.
+   */
+  public static fromUsername(username: string, password?: SecretValue, encryptionKey?: kms.IKey): Login {
```

Comment:
Maybe move `password` and `encryptionKey` under an `options` object?

If a user wants to specify the username and generate a secret with a specific key he needs to do `fromUsername('superadmin', undefined, myKey)`...

Reply:
Yeah, I was tempted to do this myself; good call.